ui/components/app/reveal-SRP-modal/reveal-SRP-modal.js

```diff
@@ -0,0 +1,97 @@
+import PropTypes from 'prop-types';
+import React, { useCallback, useState } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  Display,
+  TextVariant,
+  FontWeight,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getSeedPhrase } from '../../../store/actions';
+import {
+  Box,
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  ButtonPrimary,
+  ButtonSecondary,
+  FormTextField,
+} from '../../component-library';
+
+export default function RevealSRPModal({
+  setSecretRecoveryPhrase,
+  onClose,
+  isOpen,
+}) {
+  const t = useI18nContext();
+
+  const [password, setPassword] = useState('');
+
+  // Password error would result from appState
+  const warning = useSelector((state) => state.appState.warning);
+
+  const onSubmit = useCallback(
+    async (_password) => {
+      const seedPhrase = await getSeedPhrase(_password);
+      setSecretRecoveryPhrase(seedPhrase);
+    },
+    [setSecretRecoveryPhrase],
+  );
+
+  const handleKeyPress = useCallback(
```

Comment:
Instead of this, can we wrap the text field in a `<form>` tag?  i.e.

`<form onSubmit={(e) =>{ e.preventDefault();  onSubmit();  }>....`

Reply:
done in 506c02cefd